catalyst_voices_packages/catalyst_cardano_serialization/lib/src/scripts.dart

```diff
@@ -0,0 +1,497 @@
+  ///  scripts to include in the [ScriptAll].
+  /// - 'any': Creates a ScriptAny instance. The JSON object should have a
+  /// 'scripts' field that contains a list of JSON objects representing the
+  /// scripts to include in the [ScriptAny].
+  /// - 'atLeast': Creates a ScriptNOfK instance. The JSON object should have a
+  /// 'required' field that contains the number of scripts that must be
+  /// satisfied, and a 'scripts' field that contains a list of JSON objects
+  /// representing the scripts to include in the [ScriptNOfK].
+  /// - 'before': Creates an InvalidBefore instance. The JSON object should have
+  /// a 'slot' field that contains the slot number to use in the
+  /// [InvalidBefore].
+  /// - 'after': Creates an InvalidAfter instance. The JSON object should have a
+  /// 'slot' field that contains the slot number to use in the [InvalidAfter].
+  ///
+  /// If the JSON object contains an unsupported script type, an ArgumentError
+  /// is thrown.
+  static NativeScript fromJSON(Map<String, dynamic> json) {
+    return switch (json['type']) {
+      'sig' =>
+        ScriptPubkey(Uint8List.fromList(hex.decode(json['keyHash'] as String))),
+      'all' => ScriptAll(
+          (json['scripts'] as List<Map<String, dynamic>>)
+              .map<NativeScript>(fromJSON)
+              .toList(),
+        ),
+      'any' => ScriptAny(
+          (json['scripts'] as List<Map<String, dynamic>>)
+              .map<NativeScript>(fromJSON)
+              .toList(),
+        ),
+      'atLeast' => ScriptNOfK(
+          json['required'] as int,
+          (json['scripts'] as List<Map<String, dynamic>>)
+              .map<NativeScript>(fromJSON)
+              .toList(),
+        ),
+      'before' => InvalidAfter(json['slot'] as int),
+      'after' => InvalidBefore(json['slot'] as int),
+      _ => throw ArgumentError('Unknown script type: ${json['type']}')
+    };
+  }
+
+  static void _invalidCborError(CborValue value) =>
+      throw ArgumentError.value(value, 'value', 'Invalid NativeScript cbor');
+
+  /// Checks if a CBOR value is a valid generic native script.
+  static void _checkGenericNativeScriptValidity(CborValue value) {
+    if (value is! CborList || value.length != 2 || value[1] is! CborBytes) {
+      _invalidCborError(value);
+    }
+  }
+
+  /// Checks if a CBOR value is a valid list-based native script.
+  static void _checkListNativeScriptValidity(CborValue value) {
+    if (value is! CborList || value.length != 2 || value[1] is! CborList) {
+      _invalidCborError(value);
+    }
+  }
+
+  /// Checks if a CBOR value is a valid bounded native script.
+  static void _checkBoundedNativeScriptValidity(CborValue value) {
+    if (value is! CborList || value.length != 2 || value[1] is! CborSmallInt) {
+      _invalidCborError(value);
+    }
+  }
+}
+
+/// Class representing a public key based native script.
+class ScriptPubkey extends NativeScript {
+  /// Public key hash.
+  final Uint8List addrKeyhash;
```

Comment:
```suggestion
  final Uint8List addrKeyHash;
```

Consider to capitalize separate words separately.

Reply:
Would it be possible to use here `Ed25519PublicKeyHash` instead of Uint8List?